src/payment.rs

```diff
@@ -0,0 +1,106 @@
+use gstd::cmp::Ordering;
+use gstd::ops::{Add, AddAssign};
+use gstd::TypeInfo;
+// use gstd::debug;
+
+use codec::{Decode, Encode};
+
+use crate::Compound;
+
+#[derive(Debug, Decode, Encode, Copy, Clone, TypeInfo, Default)]
```

Comment:
It is to heavy struct to be `Copy`.
```suggestion
#[derive(Debug, Decode, Encode, Clone, TypeInfo, Default)]
```